### DISCUSSION TITLE
Cascading donations contract

### DISCUSSION THREAD

**Discussion by L-Zuluaga** [stellar/sorobanathon#80] [General]
The idea behind this contract is to cover different donations schemes, allowing to cascade the donations down to the recipients and recipients of recipients in an automated way with a single donation. I included an example using a repository donation scheme, where we donate to a project, and it automatically cascades the donation into its dependencies.
![image](https://user-images.githubusercontent.com/32212593/204314237-1be74c28-e766-4bc0-995d-d4ab2c2e2619.png)

It is a cool example of how to work with cross contract calls using only 1 contract.

The contract code and the tests are available in this [GitHub repository](https://github.com/kommitters/soroban_smart_contracts/tree/main/cascading_donations).

We went further and created a [Gitbook](https://kommitters.gitbook.io/soroban-examples/cascading-donations-contract/cascading-donations) explaining the entire contract workflow through an example.

I would like to thank `dmytro_stellar` (that's his name in discord 😅) who helped me to solve some doubts that have arisen while working with this contract.

Cheers,
Lorenzo,
kommit - [www.kommit.co](http://www.kommit.co/)

**Comment by achallagundla** [comment]
Heya @L-Zuluaga  ,
Congratulations! Your contribution to Sorobanathon: First Light has been accepted! 🎉
Your submission may qualify for a reward of 1000 XLM! One last step: to receive that reward, please fill out this [form](https://stellarform.typeform.com/Sorobanathon) completely and accurately.

Please let us know if you have any questions. (You can also message me on discord @ anuxhya)

All the best,
Anuhya

PS: No need to submit the form if you already have, just update me on discord if your wallet address has changed.

**Reply by L-Zuluaga** [reply]
Hi @achallagundla Thanks for the information, I wrote you a message in discord because we already filled out the form for a grant we won some time ago. 
This is my discord username: **Lorenzo Zuluaga#1011**

Cheers,
Lorenzo
